Scripts/Player_Scripts/player.gd

```diff
@@ -35,8 +46,29 @@ func handle_movement():
 	var input_dir := Input.get_vector("move_left", "move_right", "move_up", "move_down")
 	var direction := (transform.basis * Vector3(input_dir.x, 0, input_dir.y)).normalized()
 	if direction:
-		velocity.x = direction.x * speed
-		velocity.z = direction.z * speed
+		velocity.x = direction.x * current_move_speed
+		velocity.z = direction.z * current_move_speed
 	else:
-		velocity.x = move_toward(velocity.x, 0, speed)
-		velocity.z = move_toward(velocity.z, 0, speed)
+		velocity.x = move_toward(velocity.x, 0, current_move_speed)
+		velocity.z = move_toward(velocity.z, 0, current_move_speed)
+
+func handle_joystick_rotation():
+	var rotation_input = Input.get_vector("look_up", "look_down", "look_left", "look_right")
+	rotation_degrees.y -= rotation_input.y * joystick_sensitivity
+	rotation_degrees.x -= rotation_input.x * joystick_sensitivity
+
+func crouch(delta: float):
+	if Input.is_action_just_pressed("crouch") && (b_is_crouching == false):
+		b_is_crouching = true
+		current_move_speed = crouch_walk_speed
+		$AnimationPlayer.play("crouch")
+		print($CollisionShape3D.shape.height)
+	elif Input.is_action_just_pressed("crouch") && (b_is_crouching == true):
+		$RayCast3D.force_raycast_update()
```

Comment:
Might be worth considering if we really need the forced update, I don't think our current use case will matter if we are a few pixels out.
This would need testing but we could save some performance if we could do away with it.